compiler/src/iree/compiler/Codegen/Common/Passes.td

```diff
@@ -470,70 +453,71 @@ def SplitFullPartialTransfer :
   ];
 }
 
-def TensorToVectorVectorizePad :
+def TensorToVectorVectorizePadPass :
     InterfacePass<"iree-codegen-vectorize-tensor-pad", "mlir::FunctionOpInterface"> {
   let summary = "Vectorize a very specific form of tensor.pad with "
                 "control flows";
-  let constructor =
-      "mlir::iree_compiler::createVectorizePadPass()";
 }
 
-def TestExecutablePreprocessing :
+def TestExecutablePreprocessingPass :
     Pass<"iree-codegen-test-executable-preprocessing", ""> {
   let summary = "Tests iree-hal-preprocess-executables-with behavior.";
-  let constructor = "mlir::iree_compiler::createTestExecutablePreprocessingPass()";
 }
 
-def TestPartitionableLoopsInterface :
+def TestPartitionableLoopsInterfacePass :
     Pass<"iree-codegen-test-partitionable-loops-interface", ""> {
   let summary = "Test the PartitionableLoopsInterface";
-  let constructor = "mlir::iree_compiler::createTestPartitionableLoopsInterfacePass()";
 }
 
-def TileAndDistributeToWorkgroups :
+def TileAndDistributeToWorkgroupsPass :
     InterfacePass<"iree-codegen-tile-and-distribute-to-workgroups", "mlir::FunctionOpInterface"> {
   let summary = "Tile and distribute operations to workgroups";
-  let constructor = "mlir::iree_compiler::createTileAndDistributeToWorkgroupsPass()";
   let options = [
     Option<"maxWorkgroupParallelDims", "max-workgroup-parallel-dims", "int32_t",
-      /*default=*/ "",
+      /*default=*/ "kNumMaxParallelDims",
       "Maximum number of dims to distribute workgroups across.">,
-    Option<"distributionMethod", "distribution-method", "int32_t",
-      /*default=*/ "0",
-      "Pick the distribution method">
+    Option<"distributionMethod", "distribution-method", "linalg::DistributionMethod",
+      /*default=*/ "linalg::DistributionMethod::Cyclic",
+      "Pick the distribution method. See linalg::DistributionMethod for details",
+      [{::llvm::cl::values(
+            clEnumValN(linalg::DistributionMethod::Cyclic,
+                       "0", "Use Cyclic strategy"),
+            clEnumValN(linalg::DistributionMethod::CyclicNumProcsGeNumIters,
+                       "1", "Use CyclicNumProcGeNumIter strategy"),
+            clEnumValN(linalg::DistributionMethod::CyclicNumProcsEqNumIters,
+                       "2", "Use CyclicNumProcGeNumIter strategy"),
+            clEnumValN(linalg::DistributionMethod::None,
+                       "3", "Use None strategy")
+        )}]>,
   ];
 }
```

Comment:
This is the non-trivial part for `TileAndDistributeToWorkgroupsPass`. @MaheshRavishankar please take a look.

Reply:
A typo is fixed in https://github.com/iree-org/iree/pull/18166/commits/6db4e408abe2bf76b78257027c7cc342d720cbf6